doc/parallel.rst

```diff
@@ -69,19 +69,36 @@ Verifies that MPI, NEURON, and Python are all working together.
-#. Output from the simulation (either to ``stdout`` or ``stderr``) is communicated back to ``MPIBackend``, where it will be printed to the console. Typical output at this point would be simulation progress messages as well as any MPI warnings/errors during the simulation.
-#. Once the simulation has completed, the child process with rank 0 (in ``MPISimulation.run()``) sends a signal to ``MPIBackend`` that the simulation has completed and simulation data will be written to ``stderr``.  The data is pickled and base64 encoded before it is written to ``stderr`` in ``MPISimulation._write_data_stderr()``. No other output (e.g. raised exceptions) can go to ``stderr`` during this step.
-#. At this point, the child process with rank 0 (the only rank with complete simulation results) will send another signal that includes the expected length of the pickled and encoded data (in bytes) to ``stderr`` following the data written in the previous step. ``MPIBackend`` will use this signal to know that data transfer has completed and it will verify the length of data it receives, printing a ``UserWarning`` if the lengths don't match.
-
-It is important that ``MPISimulation`` uses the ``flush()`` method after each signal to ensure that the signal will immediately be available for reading by ``MPIBackend`` and not buffered with other output.
-
-Tests for parallel backends utilize a special ``@pytest.mark.incremental`` decorator (defined in ``conftest.py``) that causes a test failure to skip subsequent tests in the incremental block. For example, if a test running a simple MPI simulation fails, subsequent tests that compare simulation output between different backends will be skipped. These types of failures will be marked as a failure in CI.
\ No newline at end of file
+MPI parallelization with NEURON requires that the simulation be launched with the ``nrniv`` binary
+from the command-line. The ``mpiexec`` command is used to launch multiple ``nrniv`` processes which
+communicate via MPI. This is done using ``subprocess.Popen()`` in ``MPIBackend.simulate()`` to
+launch parallel child processes (``MPISimulation``) to carry out the simulation.
+The communication sequence between ``MPIBackend`` and ``MPISimulation`` is outlined below.
+
+#. In order to pass the network to simulate from ``MPIBackend``, the child ``MPISimulation``
+   processes' ``stdin`` is used. The ready-to-use `Network` object is pickled and base64 encoded
+   before being written to the child processes' ``stdin`` by way of a Queue in a non-blocking way.
+   See how it is `used in MNE-Python`_. The data is marked by start and end signals that are used
+   to extra the pickled net object. After being unpicked, the parallel simulation begins.
```

Comment:
```suggestion
   to extract the pickled net object. After being unpicked, the parallel simulation begins.
```

Reply:
Thanks @ntolley. Updated in last commit.